src/devices/src/virtio/block/persist.rs

```diff
@@ -19,17 +20,63 @@ use super::*;
 use crate::virtio::persist::VirtioDeviceState;
 use crate::virtio::{DeviceState, TYPE_BLOCK};
 
+#[derive(Clone, Copy, Versionize, PartialEq)]
+// NOTICE: Any changes to this structure require a snapshot version bump.
+pub enum CacheTypeState {
+    Unsafe,
+    Writeback,
+}
+
+impl CacheTypeState {
+    fn from_cache_type(cache_type: CacheType) -> CacheTypeState {
+        match cache_type {
+            CacheType::Unsafe => CacheTypeState::Unsafe,
+            CacheType::Writeback => CacheTypeState::Writeback,
+        }
+    }
+
+    fn create_cache_type(&self) -> CacheType {
+        match self {
+            CacheTypeState::Unsafe => CacheType::Unsafe,
+            CacheTypeState::Writeback => CacheType::Writeback,
+        }
+    }
+}
+
 #[derive(Clone, Versionize)]
 // NOTICE: Any changes to this structure require a snapshot version bump.
 pub struct BlockState {
     id: String,
     partuuid: Option<String>,
+    #[version(
+        start = 3,
```

Comment:
This is actually version 2 as it refers to the struct version, not the app version.

Reply:
Changed it.